packages/peregrine/lib/talons/OrderHistoryPage/orderHistoryContext.js

```diff
@@ -0,0 +1,31 @@
+import React, { createContext, useContext, useMemo } from 'react';
+
+import DEFAULT_OPERATIONS from './orderHistoryContext.gql';
+import { useQuery } from '@apollo/client';
+
+const OrderHistoryContext = createContext();
+
+export const OrderHistoryContextProvider = props => {
+    const operations = Object.assign({}, DEFAULT_OPERATIONS, props.operations);
+    const { getProductURLSuffixQuery } = operations;
+
+    const { data } = useQuery(getProductURLSuffixQuery, {
+        fetchPolicy: 'cache-and-network'
+    });
+
+    const storeConfig = useMemo(() => {
+        return {
+            productURLSuffix: data
+                ? data.storeConfig.product_url_suffix
+                : '.html'
+        };
+    }, [data]);
+
+    return (
+        <OrderHistoryContext.Provider value={storeConfig}>
```

Comment:
Upon further review I was wondering why this component was even necessary. Were there plans to extend it beyond just providing access to the `storeConfig`? Was the creation of this provider meant to avoid the list of N products each attempting to query for the store config?

Reply:
Exactly; I wanted to avoid the potential thundering herd of all products making this query, and the common parent was so many layers up it didn't seem ideal to prop drill. This can also lift nicely to an app-wide context when we get around to the Admin Config epic, didn't feel appropriate to take that on in this scope, but I would expect to see a solution similar to this one, just higher up in the tree.